torch/csrc/jit/codegen/cuda/expr_simplifier.h

```diff
@@ -0,0 +1,536 @@
+#pragma once
+
+#include <torch/csrc/jit/codegen/cuda/ir_all_nodes.h>
+
+// Note: [The Mathematics of Integer Arithmetic]
+//
+// Note to reader: If you just want to know what rules we are using to simplify
+// integer expressions, please go to section "Summary" in the end directly. But
+// if you want to know why we are using these rules, why these rules are correct
+// and why some others are not correct, or why some rule has some specific
+// condition, please continue reading.
+//
+// 1. Pitfalls: Integer Arithmetic vs Real Arithmetic
+// ==================================================
+//
+// We learnt arithmetic from as early as elementary school, and have been used
+// to simplify expressions using rules like a(b+c) = ab+ac, but extra care is
+// needed when dealing with integer division. Because unlike real numbers,
+// integer division can be anti-intuitive:
```

Comment:
"Integer division" is underdefined and I suggest using the more precise "truncation division", which is the default division performed by C++'s division operator (there are other types of division that could reasonably be considered "integer division," like "floor division")

Reply:
I think what I am trying to say here is, for all these division definitions, they have the following pitfall to pay attention to. I have changed the word to `integer divisions (such as Euclidean division as defined in number theory, truncation division as used by C++, and floor division as used by Python)`